netket/hilbert/boson.py

```diff
@@ -8,27 +8,25 @@
 class Boson(CustomHilbert):
     r"""Hilbert space obtained as tensor product of local bosonic states."""
 
-    def __init__(self, graph, n_max=None, n_bosons=None):
+    def __init__(self, n_max=None, N=1, n_bosons=None):
         r"""
-        Constructs a new ``Boson`` given a graph,  maximum occupation number,
+        Constructs a new ``Boson`` given a maximum occupation number, number of sites
         and total number of bosons.
 
         Args:
-           graph: Graph representation of sites.
            n_max: Maximum occupation for a site (inclusive). If None, the local occupation
                   number is unbounded.
-           n_bosons: Constraint for the number of bosons. If None, no constraint
+           N: Number of bosonic sites.
```

Comment:
n_max and N are also optional, not just n_bosons as the comments indicate.

Reply:
how can N be optional?

Reply:
It is `N=1` by default in the code at the moment.